src/solver/opamActionGraph.ml

```diff
@@ -82,8 +82,18 @@ module MakeAction (P: GenericPackage) : ACTION with type package = P.t
     | `Install p, `Install q
     | `Reinstall p, `Reinstall q
     | `Build p, `Build q
-    | `Fetch p, `Fetch q
+    | `Fetch [p], `Fetch [q]
       -> P.compare p q
+    | `Fetch pl, `Fetch ql ->
+      let rec aux comp pl ql =
+        if comp <> 0 then comp else
+        match pl, ql with
+        | [], [] -> 0
+        | [], _::_ -> -1
+        | _::_, [] -> 1
+        | p::pl, q::ql -> aux (P.compare p q) pl ql
+      in
+      aux 0 pl ql
```

Comment:
Hopefully the case shouldn't happen, but to be complete we should probably compare up to package ordering ? E.g something like 
```suggestion
      aux 0 (List.sort P.compare pl) (List.sort P.compare ql)
```
?
Well or use sets everywhere and their compare function directly, but that would probably prove extremely inconvenient everywhere else, especially for matching.